python_modules/libraries/dagster-components/dagster_components/lib/native_step_component.py

```diff
@@ -0,0 +1,63 @@
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any, Optional, Sequence
+
+from dagster import multi_asset
+from dagster._core.definitions.definitions_class import Definitions
+from dagster._core.execution.context.asset_execution_context import AssetExecutionContext
+from pydantic import BaseModel
+from typing_extensions import Self
+
+from dagster_components import Component, ComponentLoadContext
+from dagster_components.core.component import ComponentGenerateRequest
+from dagster_components.core.dsl_schema import AssetSpecModel, OpSpecBaseModel
+
+
+class NativeStepComponentSchema(BaseModel):
+    op: OpSpecBaseModel
+    assets: Optional[Sequence[AssetSpecModel]] = None
+
+
+# Possibilities
+# * op
+# * step
+# * task
+# * computation
+
+
+class NativeStepComponent(Component):
```

Comment:
I like `ComputationComponent` personally:

- op: it's more than just an op
- step: the fact that it is one step in a potentially-larger context seems somewhat irrelevant to the person implementing one of these
- task: better than the above, but doesn't feel all that precise